app/controllers/tasks_controller.rb

```diff
@@ -41,10 +48,10 @@ def destroy
   private
 
   def set_task
-    @task = Task.find_by!(id: params[:id])
+    @task = Task.find(params[:id])
   end
 
   def task_params
-    params.require(:task).permit(:project_id, :title, :description, :deadline_at)
+    params.require(:task).permit(:project_id, :title, :description, :deadline_at).merge(user_id: current_user.id)
```

Comment:
Just a note - you can use just `user` here
```suggestion
    params.require(:task).permit(:project_id, :title, :description, :deadline_at).merge(user: current_user)
```